node/task_5/main.js

```diff
@@ -0,0 +1,40 @@
+const http = require('http');
+const fs = require('fs');
+
+http.get('http://wttr.in/~kharkov', (res) => {
+    const {statusCode} = res;
+    const contentType = res.headers['content-type'];
+
+    let error;
+    if (statusCode !== 200) {
+        error = new Error('Request Failed.\n' +
+                `Status Code: ${statusCode}`);
+    } else if (!/^text\/html/.test(contentType)) {
+        error = new Error('Invalid content-type.\n' +
+                `Expected text/html but received ${contentType}`);
+    }
+    if (error) {
+        console.error(error.message);
+        res.resume();
+        return;
+    }
+
+    res.setEncoding('utf8');
+    let rawData = '';
+    res.on('data', (chunk) => {
+        rawData += chunk;
+    });
+    res.on('end', () => {
+        try {
```

Comment:
why need _try_ here? Each error will be thrown inside callback, no need to use _try ... catch_